### PR TITLE
Migrate DownloadFileFromAnchor test to Playwright

### PR DESCRIPTION
**Pull request by captainsafia**
Addresses https://github.com/dotnet/aspnetcore/issues/29739.

### CODE REVIEW

**Review by HaoK — APPROVED** [review]

File: src/Components/test/E2ETestMigration/Tests/DownloadAnchorTest.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+using System;
+using System.Threading.Tasks;
+using BasicTestApp;
+using BasicTestApp.RouterTest;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Testing;
+using Moq;
```
Comment (HaoK) [inline src/Components/test/E2ETestMigration/Tests/DownloadAnchorTest.cs:12]: Random question, does this test actually use Moq anywhere?

Comment (captainsafia) [reply src/Components/test/E2ETestMigration/Tests/DownloadAnchorTest.cs:12]: Nah, copy-paste shennagins.

Comment (captainsafia) [reply src/Components/test/E2ETestMigration/Tests/DownloadAnchorTest.cs:12]:
```suggestion
```

**Review by TanayParikh — APPROVED** [review]

**Comment by HaoK** [comment]
Finally the test app is passing on helix, feel free to merge this into my branch / or rebase and see if your tests is passing if you want.

**Pull request merged**